scripts/Connector/Connector_Molex/conn_molex_kk_396_tht_top.py

```diff
@@ -0,0 +1,234 @@
+manufacturer = 'Molex'
+orientation = 'V'
+number_of_rows = 1
+datasheet = 'https://www.molex.com/pdm_docs/sd/026604020_sd.pdf'
+
+pitch = 3.96
+drill = 1.7
+start_pos_x = 0 # Where should pin 1 be located.
+pad_to_pad_clearance = 0.8
+max_annular_ring = 1.2 # How much copper should be in y direction?
+min_annular_ring = 0.3
+
+pad_size = [pitch - pad_to_pad_clearance, drill + 2*max_annular_ring]
+if pad_size[0] - drill < 2*min_annular_ring:
+    pad_size[0] = drill + 2*min_annular_ring
+if pad_size[0] - drill > 2*max_annular_ring:
+    pad_size[0] = drill + 2*max_annular_ring
+
+pad_shape=Pad.SHAPE_OVAL
+if pad_size[1] == pad_size[0]:
+    pad_shape=Pad.SHAPE_CIRCLE
+
+eng_mpn = 'A-41791-00{n:02d}'
+new_mpn_example = '26-60-4{n:02d}0'
+
+def generate_one_footprint(pincount, configuration):
+
+    mpn = eng_mpn.format(n=pincount)
+    new_mpn = new_mpn_example.format(n=pincount)
+
+    # handle arguments
+    orientation_str = configuration['orientation_options'][orientation]
+    footprint_name = configuration['fp_name_format_string'].format(man=manufacturer,
+        series=series,
+        mpn=mpn, num_rows=number_of_rows, pins_per_row=pincount, mounting_pad = "",
+        pitch=pitch, orientation=orientation_str)
+
+    kicad_mod = Footprint(footprint_name)
+    descr_format_str = "Molex {:s}, old/engineering part number: {:s} example for new part number: {:s}, {:d} Pins ({:s}), generated with kicad-footprint-generator"
+    kicad_mod.setDescription(descr_format_str.format(series_long, mpn, new_mpn, pincount, datasheet))
+    kicad_mod.setTags(configuration['keyword_fp_string'].format(series=series,
+        orientation=orientation_str, man=manufacturer,
+        entry=configuration['entry_direction'][orientation]))
+
+    # calculate working values
+    end_pos_x = (pincount-1) * pitch
+    centre_x = (end_pos_x - start_pos_x) / 2.0
+    nudge = configuration['silk_fab_offset']
+    silk_w = configuration['silk_line_width']
+    fab_w = configuration['fab_line_width']
+
+
+    body_edge={
+        'left':start_pos_x - pitch/2,
+        'right':end_pos_x + pitch/2,
+        'bottom':5
+        }
+    body_edge['top'] = body_edge['bottom']-10.01
+
+    # create pads
+    # kicad_mod.append(Pad(number=1, type=Pad.TYPE_THT, shape=Pad.SHAPE_RECT,
+    #                     at=[0, 0], size=pad_size,
+    #                     drill=drill, layers=Pad.LAYERS_THT))
+
+    optional_pad_params = {}
+    if configuration['kicad4_compatible']:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_RECT
+    else:
+        optional_pad_params['tht_pad1_shape'] = Pad.SHAPE_ROUNDRECT
+
+    kicad_mod.append(PadArray(initial=1, start=[start_pos_x, 0],
+        x_spacing=pitch, pincount=pincount,
+        size=pad_size, drill=drill,
+        type=Pad.TYPE_THT, shape=Pad.SHAPE_OVAL, layers=Pad.LAYERS_THT,
+        **optional_pad_params))
+
+    # create fab outline
+    kicad_mod.append(RectLine(start=[body_edge['left'], body_edge['top']],\
```

Comment:
The back plate does not extend till the end so a simple rectangle will not work here.

Reply:
Fixed in 84b21113c8de2851741bfadcff3f46d2ddd73319